providers/fab/.pre-commit-config.yaml

```
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+default_stages: [pre-commit, pre-push]
+minimum_prek_version: '0.0.28'
+repos:
+  - repo: local
+    hooks:
+      - id: compile-fab-assets
+        name: Compile FAB provider assets
+        language: node
+        files: ^.*/www/
+        entry: ../../scripts/ci/prek/compile_provider_assets.py fab
+        pass_filenames: false
+        additional_dependencies: ['yarn@1.22.21']
+      - id: generate-openapi-spec-fab
+        name: Generate the FastAPI API spec for FAB
+        language: python
+        entry: ../../scripts/ci/prek/generate_openapi_spec_providers.py fab
+        pass_filenames: false
+        files: ^src/airflow/providers/fab/auth_manager/api_fastapi/.*\.py$
+      - id: check-revision-heads-map-fab
+        name: Check that the REVISION_HEADS_MAP is up-to-date
+        language: python
+        entry: ../../scripts/ci/prek/check_revision_heads_map.py
+        pass_filenames: false
+        files: >
+          (?x)
+          ^src/airflow/providers/fab/migrations/versions/.*$|
+          ^src/airflow/providers/fab/migrations/versions|
+          ^src/airflow/providers/fab/auth_manager/models/db\.py$
+      - id: prevent-deprecated-sqlalchemy-usage-fab
+        name: Prevent deprecated sqlalchemy usage
+        entry: ./scripts/ci/prek/prevent_deprecated_sqlalchemy_usage.py
```

Args... umpf still wrong path
```suggestion
        entry: ../../scripts/ci/prek/prevent_deprecated_sqlalchemy_usage.py
```